src/index.ts

```diff
@@ -76,11 +77,17 @@ function createExtensions(document, result, requestID) {
     result
   )
 
+  const optionalFieldsExtensions = optionalFieldsDirectiveExtension(
+    document,
+    result
+  )
+
   const requestLoggerExtensions = enableRequestLogging
     ? fetchLoggerRequestDone(requestID)
     : {}
 
   const extensions = {
+    ...optionalFieldsExtensions,
```

Comment:
**thought:** What should be executed first the `principalField` or the `optionalField` 🤔. I think it's safe the way you have it now since we won't be having both in the same code base ideally

Reply:
This line just adds the `optionalFields` to the extensions in the reply:

```
"data": ...,
"extensions: {
  "optionalFields": [],
  "otherExtension: [],
}

Reply:
In **Eigen** we ignore every field that's annotated with `@optionalField` regardless if it is also a `@principalField`. And ideally, we don't use them in the same query :)

Reply:
great, thanks for explaining